OpenChange/RTFHandler.m

```diff
@@ -660,22 +660,17 @@ - (void) parseStyleSheet
    } while (count != 0); 
 }
 
-//
-//
-//
-- (void) parsePicture
+- (void) parseIgnoringEverything
 {
+  unsigned int count = 1;
   // Ignore everything. But we cannot parse it blindly because it could have
   // binary data with '}' and '{' bytes, so disasters can happen and they will
-  unsigned int count = 0;
-  const char *cw;
-
   do
     {
       if (*_bytes == '\\')
         {
           unsigned int binary_size, len = 0, cw_len;
-          cw = [self parseControlWord: &len];
+          const char *cw = [self parseControlWord: &len];
```

Comment:
IMHO, the variables should have the smallest possible scope allowed by the language. So normally that means to declare them when they are initialized (and this normally means that should be the first time they are used).

Besides the declaration at the start of a function is just an old style (actually the old c89 require declaration at start of the block, not function) which does not have any advantage.

Both standards C and C++ are evolving with this path so I think this is the best way, sogo-openchange doesn't have a strict coding style so whenever I have to decide about something I use what I think is best.